### PR TITLE
feat(chat): support model `reference` instead of `id` into `conversationId` (Issue #1811)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**
support model `reference` instead of `id` into `conversationId` 

Issues:

- Issue #1811

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1822.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/498508)

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1822.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/share/share.epics.ts [hunk 1]
```diff
@@ -958,7 +958,7 @@ const deleteOrRenameSharedFolderEpic: AppEpic = (action$, state$) =>
       const isSharedFolder = folders.find(
         (folder) => folder.id === payload.folderId,
       )?.isShared;
-      const requireRevoke = payload.values ? payload.values.name : true;
+      const requireRevoke = 'values' in payload ? payload.values.name : true;
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/share/share.epics.ts]: Maybe revert this change? Previous was better because of typing i think

**Review by denys-kolomiitsev — APPROVED** [review]

File: apps/chat/src/store/share/share.epics.ts (hunk 1 above)
Comment (IlyaBondar) [reply apps/chat/src/store/share/share.epics.ts]: it was done to fix this error:
image

**Review by Alexander-Kezik — APPROVED** [review]
